lib/graphql/input_object_type.rb

```diff
@@ -75,16 +75,14 @@ def coerce_non_null_input(value)
 
       arguments.each do |input_key, input_field_defn|
         field_value = value[input_key]
-        field_value = input_field_defn.type.coerce_input(field_value)
 
-        # Try getting the default value
-        if field_value.nil?
```

Comment:
I think it did both, [`coerce_input` short-circuits](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/base_type.rb#L77) the provided `coerce_*` functions in the case of nil, so this nil check serves a double purpose: 

- if the value was initially `nil` (on a key miss), it would be nil here because of that short-circuit 
- if there was a value but it failed coercion, the coerce function should return nil

Looks to me like your code covers both those cases